Ginger/Ginger/Actions/ActionEditPages/ActExcelEditPage.xaml.cs

```diff
@@ -96,21 +93,62 @@ private void BrowseExcelButton_Click(object sender, RoutedEventArgs e)
             }) is string fileName)
             {
                 ExcelFileNameTextBox.ValueTextBox.Text = fileName;
+                SheetsList = null;
+                // Empty the list in the UI before Creating a new list
+                GingerCore.General.FillComboFromList(SheetNamComboBox, SheetsList);
                 FillSheetCombo();
             }
         }
 
-        private void FillSheetCombo()
+        private async Task FillSheetCombo(bool initialLoad = false )
         {
             ContextProcessInputValueForDriver();
-            //Move code to ExcelFunction no in Act...
-            List<string> SheetsList = mExcelOperations.GetSheets(mAct.CalculatedFileName);
-            if (SheetsList == null || SheetsList.Count == 0)
+            if (SheetsList == null || !SheetsList.Any())
             {
-                Reporter.ToUser(eUserMsgKey.ExcelInvalidFieldData);
-                return;
+                DisableSheetNameComboBox();
+                xLoader.Visibility = Visibility.Visible;
+                mExcelOperations.Dispose();
+                await Task.Run(() =>
+                {
+                    try
+                    {
+                        SheetsList = mExcelOperations.GetSheets(mAct.CalculatedFileName);
+
+                        if (SheetsList == null || SheetsList.Count == 0)
+                        {
+                            ShowErrorResponse();
+                            return;
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        ShowErrorResponse();
+                    }
+                    finally
+                    {
+                        Dispatcher.Invoke(() =>
+                        {
```

Comment:
Please move this code from Finally and add it to catch part

Reply:
What the code inside finally block does is, it disables the loader and enables the Sheetname drop down. If this is moved to the catch block, then only when the read operation fails , is when the loader is stopped. if the Sheet name is read successfully, the loader will still be visible